optuna/trial.py

```diff
@@ -372,14 +372,17 @@ def should_prune(self, step):
 
         Args:
             step:
-                Step of the trial (e.g., epoch of neural network training).
+                Deprecated: Step of the trial (e.g., epoch of neural network training).
 
         Returns:
             A boolean value. If :obj:`True`, the trial should be pruned. Otherwise, the trial will
             be continued.
         """
-
-        # TODO(akiba): remove `step` argument
+        if step is None:
+            warnings.warn(
```

Comment:
This `warn` would be for `step is not None`.

Reply:
You mean this deprecation warning should be exported if `step is not None` , right?

I guess `step` argument will remove in the future from this todo comment. So I make the use of step argument deprecated.
https://github.com/pfnet/optuna/pull/398/files#diff-815bd38671f30e415efa796835df638fL382

Reply:
I think `step = max(...)` is the expected logic that should happen if `step is None`, and `warnings.warn(...)` should be in the deprecated path (i.e., `step is not None`). Both `step = max(...)` and `warnings.warn(...)` are in the expected path in the current implementation.

What I was thinking of is as follows: 
```python
if step is None:
    # expected path
    step = max(self.storage.get_trial(self._trial_id).intermediate_values.keys())
else:
    # deprecated path
    warnings.warn('The use of `step` argument is deprecated. '
                  'You can omit to pass this parameter.', DeprecationWarning)
```

What do you think?

Reply:
Sorry, I understand. You are completely right :pray:

Reply:
Thanks ! I fixed at 25d4f2c